compiler/ir/serialization.common/src/org/jetbrains/kotlin/backend/common/serialization/DescriptorByIdSignatureFinder.kt

```diff
@@ -22,7 +22,9 @@ class DescriptorByIdSignatureFinder(
     private val lookupMode: LookupMode
 ) {
     init {
-        assert(lookupMode != LookupMode.MODULE_ONLY || moduleDescriptor is ModuleDescriptorImpl)
+        assert(lookupMode != LookupMode.MODULE_ONLY || moduleDescriptor is ModuleDescriptorImpl) {
+            "DescriptorByIdSignatureFinder.init"
```

Comment:
Probably this was for debug and is no longer needed?

Reply:
Yes, exactly. I cannot recall the specific circumstances, but some part of the test framework/error reporting stack swallowed this origin and just reported a non-descript error without this precise location.

Can we come up with something more fitting than `"DescriptorByIdSignatureFinder.init"` for the failing case, I think it will improve debugability.

Reply:
Sounds weird. :) In any case, let's either drop it, or replace with some meaningful message, e.g. `"Incorrect lookup mode $lookupMode for $moduleDescriptor"`

Reply:
Done.